src/backend/distributed/operations/worker_shard_copy.c

```diff
@@ -88,6 +104,37 @@ CanUseLocalCopy(uint32_t destinationNodeId)
 }
 
 
+/*
+ * SetupReplicationOriginSessionIfNotSetupAlready sets up the replication origin session
+ * if it is not already setup.
+ */
+static void
+SetupReplicationOriginSessionIfNotSetupAlready(MultiConnection *connection)
+{
+	/* Setup replication Origin if not setup already */
+	if (!SendRemoteCommand(connection, CDC_REPLICATION_ORIGIN_SESION_SETUP_CMD))
+	{
+		ReportConnectionError(connection, ERROR);
+	}
+	ForgetResults(connection);
+}
+
+
+/*
+ * ResetReplicationOriginSessionIfSetupAlready resets the replication origin session
```

Comment:
this function doesn't seem to be called?

Reply:
Removed this function.